assets/src/editor/events/ticket-assignments-manager/hooks/test/use-counts-manager.js

```diff
@@ -0,0 +1,627 @@
+			noAssignmentsMessage={ noAssignmentsMessage }
+			assignmentCounts={ updatedAssignmentCounts }
+		/>;
+	};
+	const getTestComponent = ( props ) => ( updatedProps = {} ) =>
+		<TestComponent key={ 'testComponent' } { ...props } { ...updatedProps } />;
+
+	describe( 'returns expected values when processing one ticket with many ' +
+		'dates', () => {
+		beforeEach(
+			() => {
+				testedAssignedCounts = {
+					dates: {
+						40: 1,
+						50: 1,
+						60: 2,
+					},
+					tickets: {
+						10: 3,
+						20: 1,
+					},
+				};
+				testedTicketEntities = [ allTickets[ 0 ] ];
+				// initially assigned state is empty.
+			}
+		);
+
+		it( 'it returns expected values when tickets and dates all have an ' +
+			'assignment', () => {
+			act( () => {
+				renderer = TestRenderer.create( getTestComponent( {
+					dateEntities: allDates,
+					ticketEntities: testedTicketEntities,
+					assignedCounts: testedAssignedCounts,
+					assignedState: testedAssignedState,
+				} )() );
+			} );
+			const testInstance = renderer.root.findByType( 'div' );
+			expect( testInstance.props.hasNoAssignments ).toBe( false );
+			expect( testInstance.props.noAssignmentsMessage )
+				.toEqual( '' );
+			expect( testInstance.props.assignmentCounts ).toEqual( testedAssignedCounts );
+		} );
+		it( 'returns expected values when incoming state for ticket has no ' +
+			'dates assigned and then has dates assigned', () => {
+			testedTicketEntities = [ allTickets[ 2 ] ];
+			testedAssignedCounts.tickets[ 30 ] = 0;
+			const Test = getTestComponent(
+				{
+					dateEntities: allDates,
+					ticketEntities: testedTicketEntities,
+					assignedCounts: testedAssignedCounts,
+					assignedState: testedAssignedState,
+				}
+			);
+			act( () => {
+				renderer = TestRenderer.create( <Test /> );
+			} );
+			const testInstance = renderer.root;
+			const getDiv = () => testInstance.findByType( 'div' );
+			let div = getDiv();
+			expect( div.props.hasNoAssignments ).toBe( true );
+			expect( div.props.noAssignmentsMessage )
+				.toEqual( expectedMessages.singleTicket );
+			expect( div.props.assignmentCounts ).toEqual( testedAssignedCounts );
+
+			// should update when ticket is assigned a date
+			const updatedAssignedState = {
+				assigned: { 40: [ 30 ] },
+				removed: {},
+			};
+			act( () => {
+				renderer.update(
+					<Test assignedState={ updatedAssignedState } />
+				);
+			} );
+			div = getDiv();
+			expect( div.props.hasNoAssignments ).toBe( false );
+			expect( div.props.noAssignmentsMessage ).toEqual( '' );
+			expect( div.props.assignmentCounts ).toEqual(
+				{ ...testedAssignedCounts,
+					dates: {
+						...testedAssignedCounts.dates,
+						40: 2,
+					},
+					tickets: {
+						...testedAssignedCounts.tickets,
+						30: 1,
+					},
+				},
+			);
+			// original assigned counts should not have changed
+			expect( testedAssignedCounts.dates[ 40 ] ).toBe( 1 );
+			expect( testedAssignedCounts.tickets[ 30 ] ).toBe( 0 );
```

Comment:
One of the issues I discovered in the latest round of changes is that the original assigned counts coming into the custom hook were getting mutated.  This is primarily because I was treating the spread operator as doing a deep clone (🤦‍♂ ).  So I added this expectation (in all tests) which both exposed it, and also verifies that the original assignment counts don't get mutated.